.github/workflows/action.yml

```
@@ -49,9 +49,37 @@ jobs:
         check-path: 'examples/cpp/'
         exclude-regex: 'examples/cpp/build'
 
-  build-cpp:
+  version-manager:
     needs: [quick-tests]
     runs-on: ubuntu-latest
+    outputs:
+      head_hash: ${{ steps.head-hash.outputs.head_hash }}
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Rust setup
+      uses: actions-rs/toolchain@v1.0.6
+      with:
+        toolchain: nightly
+        override: true
+    - name: Modify version if release-tag
+      if: startsWith(github.ref, 'refs/tags/')
+      run: |
+        cargo install cargo-bump --force \
+        && cargo bump ${{ github.ref_name }}
+    - name: Automatic commit for version upgrade
+      if: startsWith(github.ref, 'refs/tags/')
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        branch: master
```

this should use the branch name of the tag over master directly:
```suggestion
        # Pseudocode
        branch: steps.extract_branch.outputs.branch
```

The idea is to support versions released that are not master based only, the use case is patch and minor versions that are released from branches over master, or any release that is not master based.

---

Then we need to change the trigger method, currently we do all actions based on tag, from tag we can't get the branch.

```
Or it can create it's own branch 
    # Optional. Local and remote branch name where commit is going to be pushed
    #  to. Defaults to the current branch.
    #  You might need to set `create_branch: true` if the branch does not exist.
    branch: feature-123
```

branch: commit_name_version,

What is the point? To not have minor releases commits, will it replace the shown last release on the page?